Safety/Src/PWM.cpp

```diff
@@ -47,10 +51,109 @@ void PWMChannel::set(uint8_t channel, uint8_t percent)
     }
 
     PWMPinConfig currentChannel = PWM_CONFIG[channel];
-    uint32_t prescaler = (static_cast<TIM_HandleTypeDef *>(currentChannel.timer))->Init.Prescaler;
-	uint32_t us = ((percent * (PWMChannel::max_signal - PWMChannel::min_signal)) / 100 + PWMChannel::min_signal);
-    uint32_t periodTicks = (static_cast<TIM_HandleTypeDef *>(currentChannel.timer))->Init.Period;
-	uint32_t ticks = static_cast<uint32_t>((static_cast<float>(us) / static_cast<float>(pwmPeriod)) * static_cast<float>(periodTicks));
-    __HAL_TIM_SET_COMPARE((TIM_HandleTypeDef *) currentChannel.timer, currentChannel.timer_channel, (uint32_t) ticks);
 
+    if (currentChannel.isUsingDshot)
+    {
+        //prepare the buffers
+        dshotPrepareDMABuffer(currentChannel.dshotDMABuffer, percent);
+        //enable dma
+        dshotStartDMA(currentChannel);
+        //enable TIM dma requests?
+        dshotEnableDMARequests(currentChannel);
+    }
+
+    else
+    {
+        uint32_t prescaler = (static_cast<TIM_HandleTypeDef *>(currentChannel.timer))->Init.Prescaler;
+	    uint32_t us = ((percent * (PWMChannel::max_signal - PWMChannel::min_signal)) / 100 + PWMChannel::min_signal);
+        uint32_t periodTicks = (static_cast<TIM_HandleTypeDef *>(currentChannel.timer))->Init.Period;
+	    uint32_t ticks = static_cast<uint32_t>((static_cast<float>(us) / static_cast<float>(pwmPeriod)) * static_cast<float>(periodTicks));
+        __HAL_TIM_SET_COMPARE((TIM_HandleTypeDef *) currentChannel.timer, currentChannel.timer_channel, (uint32_t) ticks);
+    }
+    
+}
+
+void PWMChannel::dshotPrepareDMABuffer(uint32_t * dmaBuffer, uint8_t throttlePercentage)
+{
+    uint16_t frame = dshotPrepareFrame(throttlePercentage, false);
+
+    for (int i = 0; i < DSHOT_DATA_FRAME_LEN; i++)
+    {
+        dmaBuffer[i] = (frame & 0x8000 ? DSHOT_BIT_1 : DSHOT_BIT_0); //using the frame, populate the buffer with the duty cycle value corresponding to a 1 and 0
+
+        frame <<= 1;
```

Comment:
Ik what you're doing here, but a quick comment will help the youngins and others who read this understand better :)